File: trunk/drools/drools-examples/src/main/org/drools/examples/state/State.java
package org.drools.examples.state;

public class State
{
    private String name;

    private String state;

    public State(String name)
    {
        this.name = name;
        this.state = "NOTRUN";
    }

    public String getName()
    {
        return this.name;
    }

    public String getState()
    {
        return this.state;
    }

    public String setState(String state)
    {
        return this.state = state;
    }

    public boolean inState(String name, String state)
    {
        boolean b = this.name.equals( name ) && this.state.equals( state );
        return b;
    }

    public String toString()
    {
        return this.name + "[" + this.state + "]";
    }
}

File: drools/drools-dotnet/src/semantics.dotnet.tests/examples/state/State.cs
using System;

namespace org.drools.semantics.dotnet.tests.examples.state
{
	public class State
	{
		private string _name;
		private Status _status;

		public State(string name)
		{
			_name = name;
			_status = Status.NotRun;
		}

		public virtual string Name
		{
			get{ return _name; }
		}
		
		public virtual Status Status
		{
			get { return _status; }
			set { _status = value; }
		}
		
		public override string ToString()
		{
			return "[" + _name + ":" + _status + "]";
		}
	}
}